torch/nn/parallel/distributed.py

```diff
@@ -622,7 +622,9 @@ def _register_comm_hook(self, state: object, hook: callable):
                             feedback in gradient compression, peers to communicate with
                             next in GossipGrad etc.
             hook (callable): is defined as:
-                             hook(state: object, bucket: dist.GradBucket) -> torch.futures.Future:
+                             hook(state: object, bucket: dist._GradBucket) -> torch.futures.Future:
+                             or
+                             hook(state: object, bucket: dist._GradBucket) -> torch._C.Future:
```

Comment:
My feeling is that we should probably have only one of these, although I'll let @mrshenli confirm this.

Reply:
Agree, we should can just say it returns `torch.futures.Future` here, as this is a requirement for user code. 

Then in `get_future` API doc, we highlight that it does not return `torch.future.Future` and describe all the differences there. And also mention if the application would like to return `get_future()`'s future from `hook(...)`, what they need to be aware.